xbmc/cores/AudioEngine/CMakeLists.txt

```diff
@@ -131,6 +134,9 @@ endif()
 
 core_add_library(audioengine)
 target_include_directories(${CORE_LIBRARY} PRIVATE ${CMAKE_CURRENT_SOURCE_DIR})
+if(OVERRIDES_CPP)
+  target_compile_definitions(${CORE_LIBRARY} PRIVATE -DPLATFORM_OVERRIDE)
```

Comment:
The define PLATFORM_OVERRIDE has to be unique, maybe PLATFORM_OVERRIDE_AE_ENCODERS. (see https://github.com/xbmc/xbmc/pull/10897). Use the same below in the ifdef.

Also OVERRIDES_CPP/OVERRIDES_H are not added to SOURCES/HEADERS and are therefore not compiled. If the overrides are not dependent on any option, this should do:

    if(OVERRIDES_CPP)
      list(APPEND SOURCES ${OVERRIDES_CPP})
      list(APPEND HEADERS ${OVERRIDES_H})
      target_compile_...

For debugging if your code is picked up, add a `message(STATUS "Use AE Encoder override")` or compile with `make VERBOSE=1` to check if the define is passed to the compiler.